### PR TITLE
sys/posix/inet: Add posix_inet and dependencies to Kconfig

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

One more for the list.  The `posix_inet` uses the `ipv6_addr` and `ipv4_addr` modules.

The `ipv*_addr` has no read dependencies (optional fmt and a header only byteorder.h) so it could just be added.

### Testing procedure

```
make clean
TEST_KCONFIG=1 make menuconfig
# search for ipv*_addr modules and add
# search for posix_inet and add
TEST_KCONFIG=1 make all
``` 
You should see the posix and ipv*_addr getting compiled.
There are no simple tests for this however... so I don't think we can add them to the `.murdock` tests.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Got another one for you @leandrolanzieri

File: sys/posix/inet/Kconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config MODULE_POSIX_INET
+    bool "Posix INet"
+    depends on MODULE_IPV4_ADDR
+    depends on MODULE_IPV6_ADDR
```
Comment (leandrolanzieri) [inline sys/posix/inet/Kconfig:11]: For what I see in the Posix Inet module, it does not really depend on ipv4 and ipv6. It just adapts its behaviour when they are present. So it seems we can drop these dependencies.

File: sys/net/network_layer/Kconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2019 HAW Hamburg
```
Comment (leandrolanzieri) [inline sys/net/network_layer/Kconfig:1]:
```suggestion
# Copyright (c) 2021 HAW Hamburg
```

File: sys/posix/Kconfig [hunk 3]
```diff
@@ -12,5 +12,6 @@ config MODULE_POSIX_HEADERS
     bool
 
 rsource "sleep/Kconfig"
+rsource "inet/Kconfig"
```
Comment (leandrolanzieri) [inline sys/posix/Kconfig:15]: Can we keep alphabetical order?

File: sys/posix/inet/Kconfig [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config MODULE_POSIX_INET
+    bool "Posix INet"
```
Comment (leandrolanzieri) [inline sys/posix/inet/Kconfig:9]:
```suggestion
    bool "Posix Inet"
```
?

File: sys/posix/inet/Kconfig (hunk 1 above)
Comment (MrKevinWeiss) [reply sys/posix/inet/Kconfig:11]: :cry: OK

File: sys/posix/inet/Kconfig (hunk 4 above)
Comment (miri64) [reply sys/posix/inet/Kconfig:9]: I would actually prefer either all lower case or all caps. Those are the two ways they are used in the [POSIX specification](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/basedefs/arpa_inet.h.html#tag_13_03) (apart from just spelling out Internet).

File: sys/posix/Kconfig (hunk 3 above)
Comment (MrKevinWeiss) [reply sys/posix/Kconfig:15]: Maybe we can add in a script... that will be the next thing I do.

File: sys/posix/inet/Kconfig (hunk 4 above)
Comment (miri64) [reply sys/posix/inet/Kconfig:9]: Or to be more precise
```suggestion
    bool "POSIX Internet Support"
```
?

**Comment by MrKevinWeiss** [comment]
I guess this one turns out pretty simple.

**Comment by MrKevinWeiss** [comment]
Done and rebased.

File: sys/posix/inet/Kconfig (hunk 4 above)
Comment (MrKevinWeiss) [reply sys/posix/inet/Kconfig]: Done

**Review by leandrolanzieri — APPROVED** [review]
LGTM, ACK!

**Pull request merged**